src/components/Courses/Courses.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+
+import { useState } from 'react';
+
+import Button from 'src/common/Button/Button';
+
+import { getCourseById } from 'src/helpers/getCourseById';
+
+import CourseCard from './components/CourseCard/CourseCard';
+import SearchBar from './components/SearchBar/SearchBar';
+import EmptyCourseList from '../EmptyCourseList/EmptyCourseList';
+import CourseInfo from '../CourseInfo/CourseInfo';
+
+import { ADD_COURSE_BTN } from 'src/constants';
+
+interface coursesProp {
+	data: object[];
+}
+
+export default function Courses(props: coursesProp) {
+	const { data } = props;
+
+	const [courses, setCourses] = useState(data);
+	const [showInfo, setShowInfo] = useState(false);
+	const [courseId, setCourseId] = useState('');
+	const components =
+		courses.length !== 0 ? (
+			courses.map((course, id) => {
+				return (
+					<CourseCard
+						course={course}
+						setShowInfo={setShowInfo}
```

Comment:
This two props can be change to only one

```suggestion
						onShowCourseClick={(id) => {
						  setShowInfo(true);
						  setCourseId(id);
						 }
```

But you can also remove `showInfo` state and show CourseInfo only if `courseId` is defined

Reply:
Could you please elaborate on this more?

Reply:
Sure, here is some pseudo code:

```ts
const [courseId, setCourseId] = useState<string | undefined>();

if (!courseId) {
  return courses.map((course) => <CourseCard course={course} onShowCourseClick={setCourseId} />
}

return (
  <CourseInfo course={courses.find((course) => course.id === courseId)} />
)
```

My suggestion is to move logic for changing active `courseId` from `CourseCard` (line 18 `setCourseId(course.id);`) to `Courses` component. By doing that Courses will be able to react to whatever happens in the CourseCard component rather than share its own responsibilities to child components.

Reply:
Thanks

Reply:
I would like to discuss this workflow if you don't mind.

Reply:
Sure, let's skip for now and discuss later. I'll review the PR tomorrow